ecs-cli/modules/cli/compose/project/project_parseV3_test.go

```diff
@@ -153,6 +154,69 @@ services:
 	}
 }
 
+func TestParseV3WithTopLevelVolume(t *testing.T) {
+	// set up file
+	composeFileString := `version: '3'
+services:
+  wordpress:
+    cap_add:
+      - ALL
+    cap_drop:
+      - NET_ADMIN
+    command: echo "hello world"
+    image: wordpress
+    entrypoint: /wordpress/entry
+    ports: ["80:80"]
+    links:
+      - mysql
+    tmpfs:
+      - "/run:rw,size=1gb"
+    read_only: true
+  mysql:
+    image: mysql
+    labels:
+      - "com.example.mysql=mysqllabel"
+      - "com.example.mysql2=anothermysql label"
+    user: mysqluser
+    working_dir: /mysqltestdir
+    privileged: true
+    volumes:
+     - /opt/data:/var/lib/mysql
+     - logging:/test/place:ro
+     - /var/lib/mysql
+    extra_hosts:
+      - "mysqlexhost:10.0.0.0"
+volumes:
+  logging:`
```

Comment:
minor nit: You could add a test case that verifies that an error is thrown if the external field is present.

Reply:
good idea! will add